model/src/main/kotlin/ScanResult.kt

```diff
@@ -64,6 +64,7 @@ data class ScanResult(
             },
             scanner = scanner,
             summary = summary.filterByPath(path),
+            additionalData = additionalData
```

Comment:
How about instead leveraging `copy` so that fields cannot be forgotten? Like

        takeIf { path.isBlank() } ?: if (provenance is RepositoryProvenance) {
            copy(
                provenance = provenance.copy(vcsInfo = provenance.vcsInfo.copy(path = path)),
                summary = summary.filterByPath(path)
            )
        } else {
            copy(summary = summary.filterByPath(path))
        }